frontend/src/components/widgets/TextInput/TextInput.tsx

```diff
@@ -100,11 +104,24 @@ class TextInput extends React.PureComponent<Props, State> {
   public render = (): React.ReactNode => {
     const { dirty, value } = this.state
     const { element, width, disabled } = this.props
-    const style = { width }
 
     return (
-      <div className="row-widget stTextInput" style={style}>
+      <div
+        className="row-widget stTextInput"
```

Comment:
I'd prefer to have a styled component here

Reply:
Can you programmatically pass a width to a styled component? The original code was `style={{ width }}`. Should I put the `position: 'relative'` in a styled component and keep the width inline?

Reply:
Yup, you can pass props into styled components and then use those props in the styling. You can take a look at `Block` for an example.